### ISSUE TITLE
Serving under a subpath location

### ISSUE THREAD

**Issue by SKJoy** [henrygd/beszel#33]
Is it possible to serve the Hub UI through a custom URL like `https://host.com/beszel/` while using the Docker instance?

My current Beszel with Docker is working great :) But accessible with the base URL `https://host.com/` only. I am using NginX reverse proxy.

**Comment by henrygd** [comment]
I don't have time to test myself currently, but this should be possible using a rewrite like in this example configuration: https://pocketbase.io/docs/going-to-production/#using-reverse-proxy

Do something like that and let me know if you can access the pocketbase instance at https://host.com/beszel/_/

If you can access PocketBase but the Beszel front end is broken at https://host.com/beszel/, please let me know what the error is. If the interface loads but data does not, please open devtools and let me know what the error says.

You may need to change your application URL in the pocketbase settings.

**Comment by SKJoy** [comment]
I understand the concept they wrote there; but that is not going to work because the URLs generated on the server side (PocketBase) won't have any idea about the location path and will always generate base URLs :( So, even if the first request works, all subsequent requests will fail with a 404. Also, in my test, even the first call fails with a 404 status.

**Comment by SKJoy** [comment]
While it is possible to fix the generated URLs using `subfilter` directive; but it has it's own side effects too; not recommended.

**Comment by henrygd** [comment]
I'll test after I've cleared out a handful of other things.

**Comment by tiltX** [comment]
I'd like to add my observations to this topic because I've just tried hosting Beszel under a subpath myself so I can use https (using Apache in my case).  I have used the following directives:

```
ProxyPass /beszel http://localhost:8090/
ProxyPass /assets http://localhost:8090/assets
ProxyPass /api http://localhost:8090/api
ProxyPass /_/ http://localhost:8090/_/
```

Accessing pocketbase (https://myhost../_/) works fine and except for the font and favicons under /static (I could proxy those as well but that clashes with other apps on my host) I am able to log into the Beszel GUI. However, it doesn't work after that. The GUI shows the titlebar but there's just a big "404" where the system list should be.

**Comment by henrygd** [comment]
@tiltX Do you see any 404 requests in the network tab? Culprit is probably hardcoded URLs in the JS. This issue has been sitting for a long time so I'll try to look into it further for the next release.

**Comment by tiltX** [comment]
@henrygd strangely there were no 404 requests other than favicon and the woff2 font. Would those prevent the display of the rest of the GUI? I've used the Firefox dev tools, maybe it doesn't show all Javascript requests? Unfortunately I've pulled down the test case again so I can't check it in another browser at the moment.

**Comment by SantiagoSchez** [comment]
Is this still in the roadmap? I am using Beszel in an environment where I don't have control of the DNS zone to add subdomains. And also I cannot forward ports, so my only solution would be to serve it from a URL like https://url/beszel

**Comment by henrygd** [comment]
Yes, this is near the top of my list so should be in the next release or two.

**Comment by gardiol** [comment]
Adding my strong desire for this feature as well :) it would be great.

**Comment by SaswatPadhi** [comment]
+1. I'd love to use this project, given how lightweight it is, but the lack of subpath support is unfortunately a deal breaker with my current setup

**Comment by ktaragorn** [comment]
If you havent yet started on it, I would like to take a crack at it maybe next weekend.

**Comment by henrygd** [comment]
@ktaragorn Thanks! I'll try to knock it out before then but I'll let you know if looks like that's not going to happen.

**Comment by ktaragorn** [comment]
Came here to check if this was still open and i see you have just started on it :P 👍

**Comment by ktaragorn** [comment]
Pushed a draft of what I have so far at https://github.com/henrygd/beszel/pull/518.

**Comment by SantiagoSchez** [comment]
Thank you very much! @henrygd

**Comment by yellowsink** [comment]
awesome stuff! this'll be really helpful for my setup :D thanks

**Comment by mariuskolleck** [comment]
Thank you thank you thank you! 🌞
Works perfectly, good start to the week!

**Comment by ktaragorn** [comment]
Works great thanks! Would be great if you could specify just a relative path such as `/beszel/` but this doesnt seem to work and needs a full url, is this a limitation of what needs to be forwarded?

**Comment by mimminou** [comment]
setting `APP_PATH` seems to work behind reverse proxies because they can trim the subpath before it hits the server, however this breaks LAN access (bypassing reverse proxies by accessing directly from localAddress:port), the server is looking for files under `APP_PATH` while they appear to still be served under `/`.
I believe this is caused by the base set to `./` in /hub/site/vite.config.ts.
I'm currently experiencing some issues running the project (undefined props causing rendering issues on react), otherwise I would have opened a PR adding conditional `base` value.

EDIT: 
I have worked around this issue by setting up my reverse proxy to intercept any incoming traffic on some port i designated over 2 paths: `/` and whatever `subpath` was chosen, then proxy route both to the hub directly, like this :

```nginx
server {
	# bound to beszel
	listen 12554; #listening port
	server_name _;
		location / {
		    proxy_pass http://localhost:12555/; # hub port
		    proxy_set_header Host $host;
		    proxy_set_header X-Real-IP $remote_addr;
		    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		    proxy_set_header X-Forwarded-Proto $scheme;

		    proxy_read_timeout 360s;
		    proxy_http_version 1.1;
		    proxy_set_header Upgrade $http_upgrade;
		    proxy_set_header Connection "upgrade";
		}

		location /beszel/ { # this is what i chose as a subpath
		    proxy_pass http://localhost:12555/;
		    proxy_set_header Host $host;
		    proxy_set_header X-Real-IP $remote_addr;
		    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		    proxy_set_header X-Forwarded-Proto $scheme;

		    proxy_read_timeout 360s;
		    proxy_http_version 1.1;
		    proxy_set_header Upgrade $http_upgrade;
		    proxy_set_header Connection "upgrade";
		}
}
```
